util/chplenv/defaults.py

```diff
@@ -0,0 +1,170 @@
+
+        # Construct path to .chplconfig file
+        if chplconfigpath:
+            self.chplconfigfile = os.path.join(chplconfigpath, '.chplconfig')
+            self.prettypath = '$CHPL_CONFIG/.chplconfig'
+        else:
+            self.chplconfigfile = os.path.join(home, '.chplconfig')
+            self.prettypath = '~/.chplconfig'
+
+        # Confirm that .chplconfig file exists
+        if not os.path.isfile(self.chplconfigfile):
+            if chplconfigpath:
+                self.warnings.append(
+                (
+                    'Warning: $CHPL_CONFIG is defined, '
+                    'but no $CHPL_CONFIG/.chplconfig file is found\n'
+                ))
+            self.chplconfigfile = None
+
+    def parse(self):
+        """ Parse .chplconfig file for acceptable env var overrides """
+
+        # Parse the .chplconfig file and populate the chplconfig dict
+        with open(self.chplconfigfile, 'r') as ccfile:
+
+            # Split lines into fields delimited by '='
+            linefields = [l.split('=') for l in ccfile.readlines()]
+
+            for linenum, fields in enumerate([lf for lf in linefields]):
+
+
+                if self.invalid_assignment(fields, linenum):
+                    continue
+
+                var, val = [f.strip() for f in fields]
+
+                if self.invalid_entry(var, val, linenum):
+                    continue
+
+                self.chplconfig[var] = val
+
+    def invalid_assignment(self, fields, linenum):
+        """ Check if formatting of fields is formatted like 'ENV = VAR' """
+
+        # Check if line is a comment (has no '=')
+        if len(fields) < 2:
+            return True
+
+        # Check if line is incorrectly formatted
+        elif len(fields) > 2:
+            line = '='.join(fields).strip('\n')
+            self.warnings.append(
+            (
+                'Warning: {0}:line {1}: Received incorrect format:\n'
+                '         > {2}\n'
+                '         Expected format is:\n'
+                '         > CHPL_VAR = VALUE\n'
+            ).format(self.prettypath, linenum, line))
+            return True
+
+        return False
+
+
+    def invalid_entry(self, var, val, linenum):
+        """ Check if entry for variable assignment is valid """
+
+        # Check if var is in the list of approved special variables
+        if var not in chplvars:
+            self.warnings.append(
+            (
+                'Warning: {0}:line {1}: '
+                '"{2}" is not an acceptable variable\n'
+            ).format(self.prettypath, linenum, var))
+            return True
+
+        # Warn about duplicate entries, but don't skip, just overwrite
+        elif var in self.chplconfig.keys():
+            self.warnings.append(
+            (
+                'Warning: {0}:line {1}: '
+                'Duplicate entry of "{2}"\n'
+            ).format(self.prettypath, linenum, var))
+            return False
```

Comment:
Fair enough, I left it in there to keep the format consistent, but it's admittedly redundant and not worth the effort.